### PR TITLE
Sec 112 enable add csrf middleware

### PR DESCRIPTION
**Pull request by mattch1**
https://companieshouse.atlassian.net/browse/SEC-112

Added middleware to the app.
Updated error controller and added a unit test.
Added html error page template.

Tested unit tests with enabled true and false.
Tested journey with enabled false.

### CODE REVIEW

**Review by mwestacott — APPROVED** [review]
LGTM but I would get another review before merging.

**Review by dlloyd1-ch — APPROVED** [review]
LGTM

**Pull request merged**
